js/models/messagecollection.js

```diff
@@ -21,6 +21,20 @@ define(function(require) {
 		model: Message,
 		comparator: function(message) {
 			return message.get('dateInt') * -1;
+		},
+		initialize: function() {
+			this.on('add', function(message) {
+				var url = OC.generateUrl('apps/mail/avatars?email={email}', {
```

Comment:
would be great to have this logic in a service of its own because then we can test this more easily ;)